### PR TITLE
Add initial local frontend

### PR DESCRIPTION
**Pull request by TaylorKNoah**
I created an initial local frontend server. It has a landing page that consists of an appbar and a cool menu icon that can be clicked but does nothing, and a 404 not found page for non-existent paths.

To run it, after you pull, cd into frontend, run `npm install` then `npm run start`. Check it out in your browser at localhost:3000.

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/node_modules
+package-lock.json
+tsconfig.json
```
Comment (kesquerra) [inline frontend/.gitignore]: I believe that this is a necessary item for this to run and doesn't contain any sensitive information.

Comment (kesquerra) [inline frontend/.gitignore]: I believe that this is a necessary item for this to run and doesn't contain any sensitive information.

**Review by kesquerra — CHANGES_REQUESTED** [review]
I wasn't able to run the frontend due to the missing tsconfig file, I don't believe there's any information that needs to be hidden within it.

File: frontend/.gitignore (hunk 1 above)
Comment (TaylorKNoah) [reply frontend/.gitignore]: Thanks for the feedback. I'll add that in! (I wasn't sure which files to leave out).

Comment (TaylorKNoah) [reply frontend/.gitignore]: I added the tsconfig.

**Review by kesquerra — APPROVED** [review]
Runs good on my end and looks really great! Nice work!

File: frontend/.gitignore (hunk 1 above)
Comment (kesquerra) [reply frontend/.gitignore]: Awesome, thanks!

**Pull request merged**